sqlglot/dialects/duckdb.py

```diff
@@ -1164,6 +1164,8 @@ class Generator(generator.Generator):
             exp.DayOfWeek: rename_func("DAYOFWEEK"),
             exp.DayOfWeekIso: rename_func("ISODOW"),
             exp.DayOfYear: rename_func("DAYOFYEAR"),
+            exp.Dayname: lambda self, e: self.func("STRFTIME", e.this, exp.Literal.string("%a")),
+            exp.Monthname: lambda self, e: self.func("STRFTIME", e.this, exp.Literal.string("%b")),
```

Comment:
We shouldn't preemptively do this, because it messes up duckdb's roundtrip and changes its semantics. We should instead encode snowflake's semantics using an arg at parse time nad only generate the strftime version if it is set.